### PR TITLE
CA-280329: Fix CrossPoolMigrateDestinationPage with Current Server showing

### PR DESCRIPTION
**Pull request by makunterry**
Major modifications:
1. Create new overload of function `CreateTargetServerFilterList` in `SelectMultipleVMDestinationPage.cs` with one more argument, type `List<VM>` ;
2. Rewrite new overload function `CreateTargetServerFilterList` in sub-class `CrossPoolMigrateDestinationPage` ;
3. Call `CreateTargetServerFilterList ` with each `VM` later in function `PopulateDataGridView` ;

Signed-off-by: Kun Ma <kun.ma@citrix.com>

### CODE REVIEW

File: XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs [hunk 1]
```diff
@@ -407,8 +410,8 @@ private void PopulateDataGridView(List<ReasoningFilter> homeserverFilters)
                             item.LoadAndWait();
                             cb.Items.Add(item);
 
-                            if ((m_selectedObject != null && m_selectedObject.opaque_ref == host.opaque_ref) ||
-                                (target != null && target.Item.opaque_ref == host.opaque_ref))
+                            if (item.Enabled && ((m_selectedObject != null && m_selectedObject.opaque_ref == host.opaque_ref) ||
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs]: Why was this change needed?

File: XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs [hunk 2]
```diff
@@ -383,6 +383,9 @@ private void PopulateDataGridView(List<ReasoningFilter> homeserverFilters)
                 {
                     var tb = new DataGridViewTextBoxCell {Value = kvp.Value.VmNameLabel, Tag = kvp.Key};
                     var cb = new DataGridViewEnableableComboBoxCell{FlatStyle = FlatStyle.Flat};
+                    List<ReasoningFilter> homeserverFilters = CreateTargetServerFilterList(
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs]: This is the only place where `CreateTargetServerFilterList`, so using the new overload here breaks the Import wizard case, where we only have an override for the old function (CreateTargetServerFilterList(IEnableableXenObjectComboBoxItem item). See ImportSelectHostPage.cs.
So by using the new overload, we will always have an empty `List<ReasoningFilter>` in the Import Wizard, whne we should have the

File: XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs [hunk 3]
```diff
@@ -383,6 +383,9 @@ private void PopulateDataGridView(List<ReasoningFilter> homeserverFilters)
                 {
                     var tb = new DataGridViewTextBoxCell {Value = kvp.Value.VmNameLabel, Tag = kvp.Key};
                     var cb = new DataGridViewEnableableComboBoxCell{FlatStyle = FlatStyle.Flat};
+                    List<ReasoningFilter> homeserverFilters = CreateTargetServerFilterList(
+                                                                selectedItem,
+                                                                new List<VM> { Connection.Resolve(new XenRef<VM>(kvp.Key)) });
```
[the remark this answers is not in the document]
Comment (makunterry) [reply XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs]: Thanks!
Will refine.

File: XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs (hunk 2 above)
Comment (makunterry) [reply XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs]: Very good point!
I will refine here. Thanks!

**Comment by makunterry** [comment]
@MihaelaStoica  and @michael2012z 
I have updated the usage of `CreateTargetServerFilterList`, please help to take a look.
Thanks!

File: XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateDestinationPage.cs [hunk 4]
```diff
@@ -148,12 +148,22 @@ protected override DelayLoadingOptionComboBoxItem CreateDelayLoadingOptionComboB
...
         {
             var filters = new List<ReasoningFilter>();
 
-            if(selectedItem != null)
+            if(selectedItem != null && vmOpaqueRefs != null && selectedVMs != null)
             {
+                List<VM> vmList = new List<VM>();
+                foreach (string _opaqRef in vmOpaqueRefs)
+                {
+                    vmList.AddRange(selectedVMs.FindAll(
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateDestinationPage.cs]: I don't think we need to call `FindAll` here, because there should be only one VM with a given opaque_ref.
So we can use `Find()` and `vmList.Add()`, e.g. 
```
foreach (string opaqRef in vmOpaqueRefs)
     vmList.Add(selectedVMs.Find(vm => vm.opaque_ref == opaqRef));
```

File: XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs (hunk 1 above)
Comment (MihaelaStoica) [reply XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs]: OK, got it.

File: XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateDestinationPage.cs (hunk 4 above)
Comment (makunterry) [reply XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateDestinationPage.cs]: Agree.

**Comment by makunterry** [comment]
@MihaelaStoica 
I have refined code of `vmList.add` part. Can you please help to take a look? 
Thanks!
